### ISSUE TITLE
Otto Getting Started Guide - Dependencies not compiling

### ISSUE THREAD

**Issue by piotrj** [hashicorp/otto#81]
Hey guys. I really love the way otto looks. Once it gets in shape it's gonna be a killer tool.

I was following the Getting Started guide and I kind of got stuck on Dependencies section.
I added the dependency section, and then I run `otto compile` (which by the way the guide does not mention. I guess people would be smart enough to figure that if you once said that after every change you need to run otto compile. But it does not hurt to say "ok now run `otto compile`") and it fails with:

```
==> Loading Appfile...
==> Fetching all Appfile dependencies...
    Fetching dependency: git::https://github.com/hashicorp/otto.git//examples/mongodb
Error compiling Appfile: Dependency 'git::https://github.com/hashicorp/otto.git//examples/mongodb' doesn't have an Otto ID yet!
```

I'm not sure if there is something that I missed but it seems like there is some intermediary step that needs to be done.

**Comment by dylangrafmyre** [comment]
I experienced the same issue. The getting started guide did not reference running `otto compile` after adding the the dependency, but I assumed it was necessary since the Appfile had be changed. 

The above error occurred for me too. However, there is an `.ottoid` file in the dependency source directory on github?

I also compile the Appfile without the dependency. Add the dependency and followed the guides instructions to `otto dev destroy` and `otto dev`. The dev vm build succeeded but of course the dependency was missing when looking up `mongodb.service.consul`.

**Comment by irubnich** [comment]
+1 also getting this

**Comment by pschultz** [comment]
otto uses github.com/hashicorp/terraform/config/module.FolderStorage. That does not copy dotfiles:

https://github.com/hashicorp/terraform/blob/0c3f2a915c/config/module/copy_dir.go#L26

**Comment by dkador** [comment]
also seeing this.

**Comment by phinze** [comment]
Thanks for the reports - looking into this now!

**Comment by houshuang** [comment]
+1 seeing this as well - was surprised to see the .ottoid file in the repo on Github.

**Comment by soplakanets** [comment]
+1 :(

P.S. As a workaround you can create a local folder, copy the examples folder there and use it like this:

```
application {

   dependency {
        source = "./examples/mongodb"
    }
}

**Comment by phinze** [comment]
Hey folks - just wanted to jump in to let you know that we're working on this - we borrow some code from Terraform that skips dotfiles as it fetches dependencies. We're working on extracting that package which will allow us to customize Otto's usage of it. We should have a fix in soon!

Also thanks for that workaround @soplakanets. :+1:

**Comment by mitchellh** [comment]
This is fixed in master! 0.1.2 will be released very shortly.
